src/graph/test/DataTest.cpp

```diff
@@ -0,0 +1,291 @@
+                          "1001->1005:(81), 1001->1006:(83)";
+        auto code = client_->execute(cmd, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+    }
+    // Get result
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "GO FROM 1001 OVER schoolmate YIELD $^[person].name,"
+                          "schoolmate.likeness, $$[person].name";
+        auto code = client_->execute(cmd, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+        std::vector<std::tuple<std::string, int64_t, std::string>> expected = {
+            {"zhangsan", 80, "lisi"},
+            {"zhangsan", 85, "xiaoming"},
+            {"zhangsan", 81, "xiaoqiang"},
+            {"zhangsan", 83, "xiaohong"},
+        };
+        ASSERT_TRUE(verifyResult(resp, expected));
+    }
+    // Get multi tags
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "INSERT EDGE friend(intimacy) VALUES "
+                          "1003->1005:(90), 1003->1006:(95)";
+        auto code = client_->execute(cmd, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+    }
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "GO FROM 1003 OVER friend YIELD $^[person].name,"
+                          "friend.intimacy, $$[person].name, $$[student].grade, $$[student].number";
+        auto code = client_->execute(cmd, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+        using valueType = std::tuple<std::string, int64_t, std::string, std::string, int64_t>;
+        std::vector<valueType> expected = {
+            {"xiaoming", 90, "xiaoqiang", "three", 20190901008},
+            {"xiaoming", 95, "xiaohong", "four", 20180901003},
+        };
+        ASSERT_TRUE(verifyResult(resp, expected));
+    }
+    // Insert wrong type
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "INSERT EDGE schoolmate(likeness) VALUES 1001->2002:(\"87\")";
+        auto code = client_->execute(cmd, resp);
+        ASSERT_NE(cpp2::ErrorCode::SUCCEEDED, code);
+    }
+    // Insert wrong num of value
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "INSERT EDGE schoolmate(likeness) "
+                          "VALUES 1001->2002:(\"hello\", \"87\")";
+        auto code = client_->execute(cmd, resp);
+        ASSERT_NE(cpp2::ErrorCode::SUCCEEDED, code);
+    }
+    // Insert wrong num of prop
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "INSERT EDGE schoolmate(name, likeness) VALUES 1001->2002:(87)";
+        auto code = client_->execute(cmd, resp);
+        ASSERT_NE(cpp2::ErrorCode::SUCCEEDED, code);
+    }
+    // Insert wrong field name
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "INSERT EDGE schoolmate(like) VALUES 1001->2002:(88)";
+        auto code = client_->execute(cmd, resp);
+        ASSERT_NE(cpp2::ErrorCode::SUCCEEDED, code);
+    }
+}
+
```

Comment:
add a TODO.
after insert two tags, try to delete one tag, and test the other one.